Automattic/woocommerce-payments
# update: payment method test mode label behavior

Fixes https://github.com/Automattic/woocommerce-payments/issues/9625

#### Changes proposed in this Pull Request









Updated the "Test mode" badge to be displayed only when the payment method is selected.
I kept (but slightly modified) the grid layout.
I attempted a flexbox implementation, but ran out of options. The main challenge was showing the BNPL messaging on a new line, together with the card icon. I was semi-successful if I placed the card icon on `position: absolute` and the `label` on `position: relative`, but it was becoming less maintainable.

Changes are mainly UI-based, so unit tests are less effective.

#### Testing instructions





- Add some products to the cart
- Go to shortcode (or block-based) checkout
- Ensure that the "Test mode" label is displayed on WooPayments payment methods only when the payment method is selected
- As a customer, navigate to the "My account > Add payment method" page
- Ensure that the "Test mode" label is displayed on WooPayments payment methods only when the payment method is selected
- As a customer, ensure you have active subscriptions (create one if you don't)
- Navigate to "My account > Subscriptions > Select an 'active' subscription > Click 'Change payment'"
- Ensure that the "Test mode" label is not displayed

-------------------


**Post merge**
